utils.js

```diff
@@ -0,0 +1,86 @@
+import path from 'node:path';
+import { Readable } from 'node:stream';
+import { finished } from 'node:stream/promises';
+
+import fs from 'fs-extra';
+import { transform } from 'lightningcss';
+import unzipper from 'unzipper';
+
+const webReleasesUrl = 'https://github.com/warp-ds/tokens/releases/download/latest/web.zip';
+const webReleasesFileName = 'web.zip';
+const tempDir = 'tmp/';
+export const outputDir = 'output/web';
+
+export const init = () => {
+  console.log('Initializing setup...');
+  fs.removeSync(tempDir);
+  fs.removeSync(outputDir);
+  fs.ensureDirSync(tempDir);
+  fs.ensureDirSync(outputDir);
+};
+
+export const downloadReleaseFile = async () => {
+  console.log('Downloading assets from tokens repo: web.zip...');
+  const res = await fetch(webReleasesUrl);
+  const destination = path.resolve(tempDir, webReleasesFileName);
+  const fileStream = fs.createWriteStream(destination, { flags: 'wx' });
+  await finished(Readable.fromWeb(res.body).pipe(fileStream));
+  console.log('Extracting asset...');
+  await unzipper.Open.file(tempDir + webReleasesFileName).then((d) => d.extract({ path: outputDir, concurrency: 5 }));
+};
+
+export const getBrandModes = () => fs.readdirSync(outputDir).filter((item) => fs.statSync(path.join(outputDir, item)).isDirectory());
+
+export const brandToName = (brand) => {
+  switch (brand) {
+    case 'finn-light':
+      return 'finn-no';
+    case 'blocket-light':
+      return 'blocket-se';
+    case 'dba-light':
+      return 'dba-dk';
+    case 'tori-light':
+      return 'tori-fi';
+  }
+};
+
+export const processHexCss = (brandMode) => {
+  let cssHex = fs.readFileSync(`./${outputDir}/${brandMode}/variables.css`, 'utf8');
+  cssHex = cssHex.replaceAll(':root ', ':root,:host ');
+  cssHex = cssHex.replaceAll('--color-', '--w-');
+  cssHex = cssHex.replaceAll('--semantic-color-', '--w-s-color-');
+  cssHex = cssHex.replaceAll('--components-', '--w-color-');
+  cssHex = cssHex.replaceAll('-default:', ':');
+
+  return cssHex;
+};
+
+export const processRGBCss = (brandMode) => {
+  // this regex converts token: rgb(255, 255, 255) to token: 255, 255, 255 which is needed for unocss to work
+  const rgbValuesRegex = /rgb\(([^)]+)\)/g;
+  // this regex removes the components rgb values from the rgb css
+  const componentsRGBRegex = /--components-.*?:\s*var\(\s*--.*\);/g;
+
+  let cssRgb = fs.readFileSync(`./${outputDir}/${brandMode}/variables-rgb.css`, 'utf8');
+  cssRgb = cssRgb.replaceAll(':root ', ':root,:host ');
+  cssRgb = cssRgb.replaceAll('--color-', '--w-rgb-');
+  cssRgb = cssRgb.replaceAll('--semantic-color-', '--w-s-rgb-');
+  cssRgb = cssRgb.replaceAll(componentsRGBRegex, '');
+  cssRgb = cssRgb.replaceAll('-default:', ':');
+  cssRgb = cssRgb.replaceAll(rgbValuesRegex, '$1');
+
+  return cssRgb;
+};
+
+export const geneateFinalCss = (css, brandMode) => {
+  const { code } = transform({
+    code: Buffer.from(css),
+    //minify: true,
```

Comment:
Since this will be coming from eik, it'll always be minified and gzipped from edge anyway.

Reply:
Could you explain what you mean that "this will be coming from eik"? I thought the CSS that is passed to this function is not coming from Eik but from @warp-ds/tokens repo and from this repo's /tokens directory.

Reply:
So lightningcss is basically taking in multiple stylesheets and generating a single stylesheet per brand, eg tori-fi.css or finn-no.css. This css will get uploaded to eik and our consumers will use it via eik in production. Since eik is basically fastify, it always minifies and gzips the file it's serving on runtime.

Reply:
Is this something new? I remember that when we were initiating the release processes for warp packages, we were instructed by the platform web team to minify our assets before publishing to NPM/Eik 🤔

Reply:
I minified now